backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Tests/Controllers/SingleRightCotrollerTest.cs

```diff
@@ -0,0 +1,102 @@
+using Altinn.AccessManagement.UI.Mocks.Utils;
+using Altinn.AccessManagement.UI.Tests.Utils;
+
+namespace Altinn.AccessManagement.UI.Tests.Controllers
+{
+    /// <summary>
+    ///     Test class for <see cref="SingleRightCotrollerTest"></see>
+    /// </summary>
+    [Collection("SingleRightCotrollerTest")]
+    public class SingleRightCotrollerTest : IClassFixture<CustomWebApplicationFactory<SingleRightController>>
+    {
+        private readonly HttpClient _client;
+        private readonly CustomWebApplicationFactory<SingleRightController> _factory;
+        /// <summary>
+        ///     Constructor setting up factory, test client and dependencies
+        /// </summary>
+        /// <param name="factory">CustomWebApplicationFactory</param>
+        public SingleRightCotrollerTest(CustomWebApplicationFactory<SingleRightController> factory)
+        {
+            _factory = factory;
+            _client = SetupUtils.GetSingleRightTestClient(factory);
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            string token = PrincipalUtil.GetAccessToken("sbl.authorization");
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        }
+
+        /// <summary>
+        ///     Method that checks at least 3 of the potential responses matches at least 3 of the responses in the mocked
+        ///     response.
+        /// </summary>
+        [Fact]
+        public async Task DelegationAccessCheck_valid_response()
+        {
+            // Arrange
+            string partyId = "50004223";
+
+            AttributeMatch attribute = new AttributeMatch
+            {
+                Id = "id",
+                Value = "value",
+            };
+            
+            List<AttributeMatch> resource = new List<AttributeMatch>();
+            resource.Add(attribute);
+            DelegationRequestDto dto = new DelegationRequestDto
+            {
+                Resource = resource,
+            };
+            
+            string jsonDto = JsonSerializer.Serialize(dto);
+            HttpContent content = new StringContent(jsonDto, Encoding.UTF8, "application/json");
+
+            // Act
+            HttpResponseMessage httpResponse = await _client.PostAsync($"accessmanagement/api/v1/singleright/checkdelegationaccesses/{partyId}", content);
+            List<DelegationAccessCheckResponse> actualResponses = await httpResponse.Content.ReadAsAsync<List<DelegationAccessCheckResponse>>();
+            int countMatches = CountMatches(actualResponses);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, httpResponse.StatusCode);
+            Assert.Equal(3, countMatches);
+        }
+        private int CountMatches(List<DelegationAccessCheckResponse> actualResponses)
+        {
+
+            int countMatches = 0;
+            foreach (AccessLevel accessLevel in Enum.GetValues(typeof(AccessLevel)))
```

Comment:
Why do you need this foreach?

Now that you've removed the randomness, you should be able to fetch the relevant expected responses and compare them to the actual responses directly.

Reply:
I fix